### PR TITLE
Finishing up mouse 

### PR DESCRIPTION
**Pull request by jasminerunity3d**
I added the files for the mouse input outside of EditorXR as stated in documents.

### CODE REVIEW

File: MouseMovementTool.cs [hunk 1]
```diff
@@ -13,16 +13,9 @@
...
     {
-        public List<ILinkedObject> linkedObjects
-        {
-            set
-            {
-                throw new NotImplementedException();
-            }
-        }
 
         public Transform cameraRig
```
Comment (AndrewTHEManeri) [inline MouseMovementTool.cs]: Where is the cameraRig info coming from?  Is an interface requiring it/filling it in?
I would expect something more like
public Transform cameraRig { private get; set; }

The getter right now will return itself which will make a stack overflow, and there is no way to put data into it.

File: MouseMovementTool.cs [hunk 2]
```diff
@@ -34,87 +27,71 @@ public Transform cameraRig
                 throw new NotImplementedException();
             }
         }
+        [SerializeField]
+        GameObject m_instructions;
```
Comment (AndrewTHEManeri) [inline MouseMovementTool.cs]: m_**I**nstructions

File: MouseMovementTool.cs [hunk 3]
```diff
@@ -34,87 +27,71 @@ public Transform cameraRig
...
+        [SerializeField]
+        float m_movementMultiplier = .1f;
```
Comment (AndrewTHEManeri) [inline MouseMovementTool.cs]: m_**M**ovementMultiplier

File: MouseMovementTool.cs [hunk 4]
```diff
@@ -34,87 +27,71 @@ public Transform cameraRig
...
+        [SerializeField]
+        GameObject m_ringPrefab;
```
Comment (AndrewTHEManeri) [inline MouseMovementTool.cs]: m_**R**ingPrefab

and so on

File: MouseMovementTool.cs [hunk 5]
```diff
@@ -34,87 +27,71 @@ public Transform cameraRig
...
-        {
-            if(instructions)
-                instructions.SetActive(!instructions.activeSelf);
-        }
-
-        private void Start()
-        {
-            GameObject instance = Instantiate(ringPrefab, cameraRig, false);
+            GameObject instance = Instantiate(m_ringPrefab, cameraRig, false);
             instance.transform.localPosition = new Vector3(0f, -0.09f, 0.18f);
```
Comment (AndrewTHEManeri) [inline MouseMovementTool.cs]: what is the significance of this position? Should either be a constant or a static readonly value

File: MouseMovementTool.cs [hunk 6]
```diff
@@ -34,87 +27,71 @@ public Transform cameraRig
...
-            ring = instance.GetComponent<Ring>();
+            m_ring = instance.GetComponent<Ring>();
         }
 
-        private void FixedUpdate()
+        void FixedUpdate()
         {
+            bool k_mouse0 = UnityEngine.Input.GetMouseButton(0);
```
Comment (AndrewTHEManeri) [inline MouseMovementTool.cs]: We have the mouse action map files above - but we're using the legacy input pattern here?
Why the switch?  In either case, we should commit to one or the other and not leave dangling files around.

File: MouseMovementTool.cs [hunk 7]
```diff
@@ -34,87 +27,71 @@ public Transform cameraRig
...
+            bool k_mouse0down = UnityEngine.Input.GetMouseButtonDown(0);
+            bool k_mouse1 = UnityEngine.Input.GetMouseButton(1);
```
Comment (AndrewTHEManeri) [inline MouseMovementTool.cs]: these are not consts so they should not start with k_

File: MouseMovementTool.cs [hunk 8]
```diff
@@ -34,87 +27,71 @@ public Transform cameraRig
...
             }
-            if (UnityEngine.Input.GetMouseButton(0))
+            if (k_mouse0)
             {
-                Vector3 forward = Vector3.Scale(cameraRig.forward, new Vector3(1f, 0f, 1f)).normalized;
-                Vector3 right = Vector3.Scale(cameraRig.right, new Vector3(1f, 0f, 1f)).normalized;
-                Vector3 delta = (UnityEngine.Input.GetAxis("Mouse X") * right + UnityEngine.Input.GetAxis("Mouse Y") * forward) * movementMultiplier;
-                //Debug.Log(delta.normalized);
-                cameraRig.position += delta;
+                const string k_mouseX = "Mouse X";
```
Comment (AndrewTHEManeri) [inline MouseMovementTool.cs]: This is a weird pedantic thing, but it _is_ our standards.

If a constant is in a function, it doesn't need the k_

File: MouseMovementTool.cs [hunk 9]
```diff
@@ -34,87 +27,71 @@ public Transform cameraRig
...
+                const string k_mouseY = "Mouse Y";
 
-                //Update trackball
-                //ball.rotation = Quaternion.AngleAxis(-UnityEngine.Input.GetAxis("Mouse X") * movementMultiplier * 10f, trackball.forward)
-                //    * Quaternion.AngleAxis(UnityEngine.Input.GetAxis("Mouse Y") * movementMultiplier * 10f, trackball.right) * ball.rotation;
+                Vector3 k_forward = Vector3.Scale(cameraRig.forward, new Vector3(1f, 0f, 1f)).normalized;
+                Vector3 k_right = Vector3.Scale(cameraRig.right, new Vector3(1f, 0f, 1f)).normalized;
```
Comment (AndrewTHEManeri) [inline MouseMovementTool.cs]: you should be able to calculate right as the perpendicular version of forward -k_forward.z, 0, k_forward.x which will handle the case of the camera rig being rotated along the roll axis more gracefully

File: MouseMovementTool.cs [hunk 10]
```diff
@@ -0,0 +1,97 @@
+#if UNITY_EDITOR
+using System;
+using System.Collections;
+using System.Collections.Generic;
+//using UnityEditor.Experimental.EditorVR.Core;
```
Comment (AndrewTHEManeri) [inline MouseMovementTool.cs]: no commented usings

File: Mousecon.cs [hunk 11]
```diff
@@ -0,0 +1,16 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class Mousecon : MonoBehaviour {
+
+	// Use this for initialization
+	void Start () {
```
Comment (AndrewTHEManeri) [inline Mousecon.cs]: Should this script still be around?

File: MouseMovementTool.cs [hunk 12]
```diff
@@ -0,0 +1,120 @@
...
+//using UnityEditor.Experimental.EditorVR.UI;
+//using UnityEditor.Experimental.EditorVR.Utilities;
+using UnityEngine;
+//using UnityEngine.InputNew;
+using UnityEngine.UI;
+
+namespace UnityEditor.Experimental.EditorVR.Tools
+{
+    sealed class MouseMovementTool : MonoBehaviour, ITool, ILocomotor, IRayVisibilitySettings,
+        /*ICustomActionMap,*/ ILinkedObject, IUsesViewerScale,
```
Comment (dunity) [inline MouseMovementTool.cs]: Remove commented interface

File: MouseMovementTool.cs [hunk 13]
```diff
@@ -0,0 +1,120 @@
...
+            set
+            {
+                throw new NotImplementedException();
+            }
+        }
+
+        public Transform cameraRig
+        {
+            private get {
+                return cameraRig;
```
Comment (dunity) [inline MouseMovementTool.cs]: Remove private access modifier.  Private modifiers should not be added; they're implied.  Match property formatting conventions for get.  i.e. : "get { return m_CameraRig; }"

File: MouseMovementTool.cs [hunk 14]
```diff
@@ -0,0 +1,120 @@
...
+        //{
+        //    throw new NotImplementedException();
+        //}
+
+        public GameObject instructions;
+        public float movementMultiplier = .1f;
+        public GameObject ringPrefab;
+        public Ring ring;
+
+        private void ToggleInstructions()
```
Comment (dunity) [inline MouseMovementTool.cs]: All instructions-related logic should be removed

File: MouseMovementTool.cs [hunk 15]
```diff
@@ -0,0 +1,120 @@
...
+                throw new NotImplementedException();
+            }
+        }
+
+        //public void ProcessInput(ActionMapInput input, ConsumeControlDelegate consumeControl)
+        //{
+        //    throw new NotImplementedException();
+        //}
+
+        public GameObject instructions;
```
Comment (dunity) [inline MouseMovementTool.cs]: Remove all public fields, replace with private fields, and public properties(where necessary)

File: MouseMovementTool.cs [hunk 16]
```diff
@@ -0,0 +1,120 @@
...
+        private void FixedUpdate()
+        {
+
+            if (UnityEngine.Input.GetMouseButtonDown(0) && UnityEngine.Input.GetMouseButtonDown(1) ||
+           UnityEngine.Input.GetMouseButton(0) && UnityEngine.Input.GetMouseButtonDown(1) ||
+           UnityEngine.Input.GetMouseButtonDown(0) && UnityEngine.Input.GetMouseButton(1))
+            {
+                ToggleInstructions();
+            }
+            if (UnityEngine.Input.GetMouseButton(0))
```
Comment (dunity) [inline MouseMovementTool.cs]: Add lines between if statements

File: Mousecon.cs [hunk 17]
```diff
@@ -0,0 +1,16 @@
...
+		
+	}
+	
+	// Update is called once per frame
+	void Update () {
+        Debug.Log(Input.GetAxis("Mouse X"));
```
Comment (dunity) [inline Mousecon.cs]: Testing-related log prints should be removed.  This class should also be deleted.

File: Ring.cs [hunk 18]
```diff
@@ -0,0 +1,111 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using TMPro;
+
+public class Ring : MonoBehaviour {
```
Comment (dunity) [inline Ring.cs]: Curly braces should reside on the line below for all classes and functions

File: Ring.cs [hunk 19]
```diff
@@ -0,0 +1,111 @@
...
+        m_initAltitude = transform.position.y;
+        m_ringMat = ringTr.GetComponent<MeshRenderer>().material;
+        m_origGradient = new Gradient();
+        m_origGradient.SetKeys(lineA.colorGradient.colorKeys, lineA.colorGradient.alphaKeys);
+        m_currentGradient = new Gradient();
+        m_currentGradient.SetKeys(lineA.colorGradient.colorKeys, lineA.colorGradient.alphaKeys);
+
+        m_lineLength = lineA.GetPosition(1).y;
+        m_lineOffset = lineA.GetPosition(0).y;
+
```
Comment (dunity) [inline Ring.cs]: Remove extra line

File: Ring.cs [hunk 20]
```diff
@@ -0,0 +1,111 @@
...
+    public Transform ringTr;
+    public TextMeshProUGUI text;
+    public TextMeshProUGUI coords;
+    public float fadeAwaySpeed = 1f;
+    public float fadeInSpeed = 5f;
+    public LineRenderer lineA, lineB;
+
+    Material m_ringMat;
+    float m_initAltitude;
```
Comment (dunity) [inline Ring.cs]: Private fields should have the character after the underscore capitalized

File: Ring.cs [hunk 21]
```diff
@@ -0,0 +1,111 @@
...
+    void Update () {
+        //Decrease intensity over time
+        m_intensity = Mathf.Max(0f, m_intensity - fadeAwaySpeed * Time.deltaTime);
+        m_intensity_core = Mathf.Max(0f, m_intensity_core - fadeAwaySpeed * Time.deltaTime);
+        m_intensity_core_up = Mathf.Max(0f, m_intensity_core_up - fadeAwaySpeed/2f * Time.deltaTime);
+        m_intensity_core_down = Mathf.Max(0f, m_intensity_core_down - fadeAwaySpeed/2f * Time.deltaTime);
+
+
+        //Update text
+        float kAltitude = transform.position.y - m_initAltitude;
```
Comment (dunity) [inline Ring.cs]: not const, remove the "k" prefix

File: Ring.cs [hunk 22]
```diff
@@ -0,0 +1,111 @@
...
+        text.text = kAltitude.ToString("0.00") + " m";
```
Comment (dunity) [inline Ring.cs]: Store "0.00" as const before concatenating here

File: Ring.shader [hunk 23]
```diff
@@ -0,0 +1,143 @@
+// Made with Amplify Shader Editor
```
Comment (dunity) [inline Ring.shader]: Remove amplify-related comments

File: Ring.shader [hunk 24]
```diff
@@ -0,0 +1,143 @@
...
+		_Front("Front", Color) = (0,0,0,0)
+		_Back("Back", Color) = (1,1,1,0)
+		_Intensity("Intensity", Float) = 1
+		[HideInInspector] _texcoord( "", 2D ) = "white" {}
+		[HideInInspector] __dirty( "", Int ) = 1
+	}
+
+	SubShader
+	{
+		Tags{ "RenderType" = "Transparent"  "Queue" = "Transparent+0" "IgnoreProjector" = "True" "IsEmissive" = "true"  }
```
Comment (dunity) [inline Ring.shader]: Remove IsEmissive

File: Ring.shader (hunk 23 above)
Comment (dunity) [reply Ring.shader]: This shader should not account for scene lighting.  It should be unlit, and only allow for textures or vertex colors for any visual definition.

File: Ring.shader [hunk 25]
```diff
@@ -0,0 +1,143 @@
+// Made with Amplify Shader Editor
+// Available at the Unity Asset Store - http://u3d.as/y3X 
+Shader "Ring"
```
Comment (dunity) [inline Ring.shader]: Should be "EditorVR/InputRing"

File: Ring.shader [hunk 26]
```diff
@@ -0,0 +1,143 @@
...
+				float3 worldPos = mul( unity_ObjectToWorld, v.vertex ).xyz;
+				fixed3 worldNormal = UnityObjectToWorldNormal( v.normal );
+				o.customPack1.xy = customInputData.uv_texcoord;
+				o.customPack1.xy = v.texcoord;
+				o.worldPos = worldPos;
+				TRANSFER_SHADOW_CASTER_NORMALOFFSET( o )
+				return o;
+			}
+			fixed4 frag( v2f IN
+			#if !defined( CAN_SKIP_VPOS )
```
Comment (dunity) [inline Ring.shader]: All spacing before #If's should be removed

File: Scripts/Core/VRView.cs [hunk 27]
```diff
@@ -342,7 +348,44 @@ void OnGUI()
...
+                        case 1:
+                            RightMouseButtonHeld = false;
+                            break;
+                        case 2:
+                            MiddleMouseButtonHeld = false;
+                            break;
+                    }
+                    break;
+            }
+
```
Comment (mtschoen-unity) [inline Scripts/Core/VRView.cs]: @AndrewTHEManeri, @dunity I found that the input manager just wasn't giving us mouse input at edit time. This was the best alternative I could think of. We should talk Rene about whether this is the case w/ the new input system.

**Pull request merged**